libs/schedulers/include/hpx/schedulers/shared_priority_queue_scheduler.hpp

```diff
@@ -275,10 +275,11 @@ namespace hpx { namespace threads { namespace policies {
 
                 // ------------------------------------------------------------
                 bool cleanup_terminated(
-                    std::size_t thread_num, bool delete_all) override
+                    std::size_t /*thread_num*/, bool delete_all) override
                 {
                     std::size_t local_num = local_thread_number();
-                    HPX_ASSERT(local_num < num_workers_);
+                    HPX_ASSERT(local_num != std::size_t(-1) &&
+                        local_num < num_workers_);
```

Comment:
Didn't you say `local_num` is _allowed_ to be `-1`?

Reply:
Ok, almost always... I didn't want to change the behaviour here, just make it clear that `thread_num` is not used. And my changed assert doesn't make much sense with `std::size_t`. In any case, @biddisco will drop these changes and redo the checks if necessary.